rust/gui-client/src-tauri/src/client/network_changes/linux.rs

```diff
@@ -17,29 +19,57 @@ pub(crate) fn check_internet() -> Result<bool> {
     Ok(true)
 }
 
-pub(crate) struct Worker {}
+pub(crate) struct Worker {
+    interval: Interval,
+}
 
 impl Worker {
     pub(crate) fn new() -> Result<Self> {
-        Ok(Self {})
+        Ok(Self {
+            interval: create_interval(),
+        })
     }
 
     pub(crate) fn close(&mut self) -> Result<()> {
         Ok(())
     }
 
-    pub(crate) async fn notified(&self) {
-        futures::future::pending().await
+    pub(crate) async fn notified(&mut self) {
+        loop {
+            self.interval.tick().await;
+            tracing::trace!("Checking for network changes");
+        }
     }
 }
 
-pub(crate) struct DnsListener {}
+pub(crate) struct DnsListener {
+    interval: Interval,
+    last_seen: Vec<IpAddr>,
+}
 
 impl DnsListener {
     pub(crate) fn new() -> Result<Self> {
-        Ok(Self {})
+        Ok(Self {
+            interval: create_interval(),
+            last_seen: crate::client::resolvers::get().unwrap_or_default(),
+        })
     }
-    pub(crate) async fn notified(&mut self) -> Result<()> {
-        futures::future::pending().await
+
+    pub(crate) async fn notified(&mut self) -> Result<Vec<IpAddr>> {
+        loop {
+            self.interval.tick().await;
+            tracing::trace!("Checking for DNS changes");
+            let new = crate::client::resolvers::get().unwrap_or_default();
+            if new != self.last_seen {
```

Comment:
Might help to have a debug log for this

```suggestion
            if new != self.last_seen {
                tracing::debug!("Found new DNS resolvers: {:?}", new);
```

Reply:
Good idea. There's already one in the platform-independent code in `gui.rs` so I added the list of resolvers there.